File: Python/202. Happy Number.py
class Solution:
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        s = set()
        while n != 1:
            ns = str(n)
            n = 0
            for c in ns:
                n += int(c)**2
            if n in s:
                return False
            s.add(n)
        return True

File: C++/202. Happy Number.cpp
class Solution {
public:
    bool isHappy(int n) {
        set<int> s;
        while(n != 1)
        {
            string ns = to_string(n);
            n = 0;
            for(int i=0;i<ns.length();i++)
                n += (ns[i]-'0')*(ns[i]-'0');
            if(s.find(n) != s.end())
                return false;
            s.insert(n);
        }
        return true;
    }
};